lib/autocomplete-provider.coffee

```diff
@@ -1,89 +1,64 @@
-_ = require 'lodash'
+iconHTML = "<img src='#{__dirname}/../static/logo.svg' style='width: 100%;'>"
 
-Config = require './config'
+regexes =
+    # pretty dodgy, adapted from http://stackoverflow.com/a/8396658
+    r: /([^\d\W]|[.])[\w.$]*$/
 
-module.exports = (kernelManager) ->
-    languageMappings = Config.getJson 'languageMappings'
-
-    selectors = _.uniq kernelManager.getAllKernelSpecs().map ({language}) ->
-        if language in languageMappings
-            return '.source.' + languageMappings[language].toLowerCase()
-        return '.source.' + language.toLowerCase()
-
-    selector = selectors.join ', '
-        .replace 'coffeescript', 'coffee'
-        .replace 'javascript', 'js'
-
-    console.log 'AutocompleteProvider: selector =', selector
-
-    return {
-        selector: selector
-        disableForSelector: '.comment'
-
-        defaultRegex: /([\w-][\.:\$]?)+$/
-        regexes:
+    # adapted from http://stackoverflow.com/q/5474008
+    python: /([^\d\W]|[\u00A0-\uFFFF])[\w.\u00A0-\uFFFF]*$/
 
-            # pretty dodgy, adapted from http://stackoverflow.com/questions/8396577/check-if-character-value-is-a-valid-r-object-name/8396658#8396658
-            r: /([^\d\W]|[.])[\w.$]*$/
+    # adapted from http://php.net/manual/en/language.variables.basics.php
+    php: /[$a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/
 
-            # this is NOT correct. using the python one until I get an alternative
-            julia: /([^\d\W]|[\u00A0-\uFFFF])[\w.!\u00A0-\uFFFF]*$/
 
-            # adapted from http://stackoverflow.com/questions/5474008/regular-expression-to-confirm-whether-a-string-is-a-valid-identifier-in-python
-            python: /([^\d\W]|[\u00A0-\uFFFF])[\w.\u00A0-\uFFFF]*$/
-
-            # adapted from http://php.net/manual/en/language.variables.basics.php
-            php: /[$a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/
+module.exports = (kernelManager) ->
+    autocompleteProvider =
+        selector: '.source'
+        disableForSelector: '.comment, .string'
 
-        # This will take priority over the default provider, which has a
-        # priority of 0.
-        # `excludeLowerPriority` will suppress any providers with a lower
-        # priority i.e. The default provider will be suppressed
+        # `excludeLowerPriority` will suppress any providers with lower priority
```

Comment:
What about `excludeLowerPriority: false` won't suppress providers with lower priority` or something similar? I think this would be clearer.